### PR TITLE
Load assets async

### PR DESCRIPTION
**Pull request by brenoc**
Breaking changes:
- Rename ResourceActions to AreaActions

Features:
- Expose Storefront service at `storefront.sdk.services.storefront`
- Load assets async
- New store: AssetStore

### CODE REVIEW

File: src/services/Storefront.js [hunk 1]
```diff
@@ -1,47 +1,47 @@
...
+const workspace = ('; ' + document.cookie).split('; vtex_workspace=').pop().split(';').shift();
 
-    return axios.put(url, data, {
-      headers: this.defaultHeaders
-    });
-  }
+export const defaultHeaders = {
+  'Authorization': `token ${token}`,
+  'x-vtex-workspace': workspace ? workspace : 'master'
+};
```
Comment (tamorim) [inline src/services/Storefront.js]: @brenoc It's a minor thing, but why not use `workspace || 'master'`instead?

File: src/utils/loadPage.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      dispatcher.actions.AreaActions.getAreaAssets({id: routeName});
+    }
+  }
+
+  let assetStoreListener = dispatcher.stores.AssetStore.listen(() => {
+    getAreaAssets();
+  });
+
+  return getAreaAssets();
+}
```
Comment (tamorim) [inline src/utils/loadPage.js]: @brenoc Can you do an overview of the new steps? If you could dive more on this file too it would be nice :)

**Comment by tamorim** [comment]
@brenoc Why expose the service too?

File: src/services/Storefront.js (hunk 1 above)
Comment (brenoc) [inline src/services/Storefront.js]: I prefer the ternary operator because you can predict the effects of it.

File: src/utils/loadPage.js (hunk 2 above)
Comment (brenoc) [inline src/utils/loadPage.js]: ok!

**Comment by brenoc** [comment]
@tamorim exposing the service let other apps to use the Storefront's API more easily.

**Comment by brenoc** [comment]
@tamorim @bellacorbo commented the code, please review :)

**Comment by brenoc** [comment]
Also refactored some things.

**Comment by tamorim** [comment]
@brenoc WIP containing the listenersMap and the detached getAreaAssets!

**Pull request merged**